base/src/test/kotlin/proguard/analysis/PartialEvaluatorErrorsTest.kt

```diff
@@ -0,0 +1,503 @@
+        }
+
+        "Variable types do not match" {
+            // Store an `a`, load an `i`, this cannot work.
+            val (programClassPool, _) = fastBuild(
+                """
+                ldc "test"
+                astore_0
+                iload_0
+                areturn
+                """.trimIndent()
+            )
+
+            shouldThrow<IllegalArgumentException> { fastEval(programClassPool, PartialEvaluator()) }
+        }
+
+        "Variable types do not match instruction" {
+            val (programClassPool, _) = fastBuild(
+                """
+                iconst_3
+                iconst_3
+                iconst_1
+                iconst_1
+                lsub
+                aload_0
+                areturn
+                """.trimIndent()
+            )
+
+            shouldThrowAny { fastEval(programClassPool, PartialEvaluator()) }
+        }
+
+
+
+        "Variable types do not match instruction - long interpreted as int" {
+            val (programClassPool, _) = fastBuild(
+                """
+                lconst_1
+                isub
+                aload_0
+                areturn
+                """.trimIndent()
+            )
+            shouldThrow<IllegalArgumentException> { fastEval(programClassPool, PartialEvaluator()) }
+        }
+
+        "anewarray" {
+            val (programClassPool, _) = fastBuild(
+                """
+                lconst_1
+                anewarray long
+                areturn
+                """.trimIndent()
+            )
+            shouldThrow<IllegalArgumentException> { fastEval(programClassPool, PartialEvaluator()) }
+        }
+
+        "getfield but field has wrong type" {
+            val (programClassPool) = fastBuild(
+                """
+                aload_0
+                getfield EmptySlot#float INT_FIELD
+                areturn
+                """.trimIndent()
+            )
+
+            shouldThrow<IllegalArgumentException> { fastEval(programClassPool, PartialEvaluator()) }
+        }
+
+        "Should complain about variable types instead of negative stack size" {
+            // The stack size will be negative in this snippet
+            // But this is cause because we used the wrong type operation
+            // Saying that lsub is not compatible with ints is more handy
+            shouldThrowAny {
+                ClassBuilder(
+                    VersionConstants.CLASS_VERSION_1_8,
+                    AccessConstants.PUBLIC,
+                    "PartialEvaluatorDummy",
+                    ClassConstants.NAME_JAVA_LANG_OBJECT
+                )
+                    .addMethod(AccessConstants.PUBLIC, "test", "()V", 50) { code ->
+                        code
+                            .iconst_3()
+                            .iconst_1()
+                            .lsub()
+                            .return_()
+                    }
+                    .programClass
+            }
+        }
```

Comment:
Does the order of complaining really matter? Both are an issue.

Reply:
removed the specific line